src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlException.cs

```diff
@@ -96,6 +96,28 @@ public override void GetObjectData(SerializationInfo si, StreamingContext contex
         /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlException.xml' path='docs/members[@name="SqlException"]/Class/*' />
         public byte Class => Errors.Count > 0 ? Errors[0].Class : default;
 
+#if NET6_0_OR_GREATER
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlException.xml' path='docs/members[@name="IsTransient"]/IsTransient/*' />
+        public override bool IsTransient
```

Comment:
I'd recommend making a shared utility of `IsTransientError(SqlException ex)` in a separate file that is currently defined in `SqlInternalConnectionTds` class and calling that from here, as well as in the `SqlInternalConnectionTds` ctor (as it is now).

Reply:
You can extract the `transientErrors` to be defined in common type and use the same here too, so driver implementation that is used internally is the same as what this API exposes.